cpp/src/arrow/acero/hash_join_node_test.cc

```diff
@@ -3449,5 +3449,98 @@ TEST(HashJoin, LARGE_MEMORY_TEST(BuildSideOver4GBVarLength)) {
                    num_batches_left * num_rows_per_batch_left * num_batches_right);
 }
 
+// GH-45334: The right side (the build side) payload column of the matching rows' are
```

Comment:
I may be able to further enhance this test to be able to place the matching key over 4GB, will try.